### PR TITLE
Add SubletrViewModel and SubletrChildViewModel

### PR DESCRIPTION
**Pull request by leeminkyu16**

### CODE REVIEW

**Review by abhaykshar — DISMISSED** [review]
now I have to rebase my changes for create listing page...

**Review by Jiay0928 — DISMISSED** [review]

File: android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.uwaterloo.subletr.infrastructure
+
+import androidx.lifecycle.ViewModel
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.Disposable
+
+open class SubletrViewModel<UiState : Any>(
```
Comment (andr-dev) [inline android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt:7]: This is good for now, but in the future it would be nice to strongly type this along with the children of the SubletrChildViewModel

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	)
+
+	data class ListingParamsAndResponse(
+		val listingParams: GetListingParams,
+		val listingsResponse: GetListingsResponse,
+	)
+
+	companion object {
+		const val LISTING_PAGE_SIZE = 5
+	}
```
Comment (andr-dev) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt:195]: We can do this in another PR, but this would be good to dump in a constants file.

**Review by andr-dev — DISMISSED** [review]
Nice!

File: android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt (hunk 1 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt:7]: This does strongly type the `UiState` type.
All `: Any` means is that the type `UiState` has to be a subclass of `Any`.
In practical terms, it means that `UiState` can't be nullable (I think).

I guess we could have stricter enforcement of typing of `SubletrChildViewModel`s if that's what you're saying.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt (hunk 2 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt:195]: Maybe, it is a constant specific to the `HomeListChildViewModel` so it might be fine there....

**Review by abhaykshar — APPROVED** [review]

File: android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt (hunk 1 above)
Comment (andr-dev) [reply android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt:7]: Nope, having a subclass of UiState works. Didn't know about those semantics, sorry.

For the childViewModels, yeah but not important right now.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt (hunk 2 above)
Comment (andr-dev) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt:195]: Yeah that could work, I was thinking more there's only going to be so many constants so might as well dump them all in one file. Whatever you think is best, save for another PR though.

**Review by andr-dev — APPROVED** [review]
Nice!

**Review by Jiay0928 — APPROVED** [review]

File: android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt (hunk 1 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt:7]: Nah, Kotlin syntax is just weird (I still think so despite having used the language for years).
